src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -27,6 +34,42 @@ internal class RPackageManager : IRPackageManager {
             return VisualComponent;
         }
 
+        public async Task<IList<RPackage>> GetInstalledPackagesAsync() {
+            if (_interactiveWorkflow.RSession == null || !_interactiveWorkflow.RSession.IsHostRunning) {
+                return new List<RPackage>();
+            }
+
+            REvaluationResult result;
+            using (IRSessionEvaluation eval = await _interactiveWorkflow.RSession.BeginEvaluationAsync()) {
+                result = await eval.InstalledPackages();
+            }
+
+            return ParsePackages(result);
+        }
+
+        public async Task<IList<RPackage>> GetAvailablePackagesAsync() {
+            if (_interactiveWorkflow.RSession == null || !_interactiveWorkflow.RSession.IsHostRunning) {
+                return new List<RPackage>();
+            }
+
+            REvaluationResult result;
+            using (IRSessionEvaluation eval = await _interactiveWorkflow.RSession.BeginEvaluationAsync()) {
+                result = await eval.AvailablePackages();
+            }
+
+            return ParsePackages(result);
+        }
+
+        private static IList<RPackage> ParsePackages(REvaluationResult result) {
+            List<RPackage> packages = new List<RPackage>();
+            foreach (var jsonProp in ((JObject)result.JsonResult).Properties()) {
```

Comment:
This looks like it could be a LINQ expression. Something like:
```cs
return (from jsonProp in (JObject)result.JsonResult).Properties()
        let jsonObj = (JObject)jsonProp.Value
        select jsonObj.ToObject<RPackage>()
       ).ToList()
```